### PR TITLE
new CLI  "beeper"  command

### PR DESCRIPTION
**Pull request by Pierre-A**
With this PR I propose a new CLI command for selectively turning on/off the beeper.
This was initiated in PR #1309 with the _beeper_off_flags_ CLI parameter which was not very friendly.

This new command, called _**beeper**_ , works as others in CLI (i.e. _feature_) and so is quite obvious (see screenshots below).
All beeping conditions are listed either with _**dump**_ or _**beeper list**_.

1 - _beeper XX_ and _beeper -XX_ respectively enable / disable the XX beeping condition.
2 - _beeper ALL_  and _beeper -ALL_ respectively enable / disable all beeping conditions.
~~3 - _beeper PREFERED_ saves the current configuration and _beeper -PREFERED_ recalls this configuration to be applied at any time.~~ **REMOVED**
4- _beeper_ lists all disabled configurations.

Comments are welcome.

beeper list fig1
beeper fig2
beeper fig3

### CODE REVIEW

File: src/main/config/config.c [hunk 1]
```diff
@@ -999,3 +999,47 @@ uint32_t featureMask(void)
...
 }
+
+void beeperOffSet(uint32_t mask)
+{
+    masterConfig.beeper_off_flags |= mask;
+}
+
+void beeperOffSetAll(uint8_t beeperCount)
+{
+    uint32_t mask = 0;
```
Comment (ledvinap) [inline src/main/config/config.c]: 'mask = (1 << beeperCount) - 1;'

File: src/main/io/beeper.c [hunk 2]
```diff
@@ -376,9 +379,9 @@ const char *beeperNameForTableIndex(int idx)
 /*
  * Returns the number of entries in the beeper-sounds table.
  */
-int beeperTableEntryCount(void)
+uint8_t beeperTableEntryCount(void)
```
Comment (ledvinap) [inline src/main/io/beeper.c]: This will produce worse code than returning int

File: src/main/config/config.c (hunk 1 above)
Comment (Pierre-A) [inline src/main/config/config.c]: of course, much more elegant.

File: src/main/io/beeper.c (hunk 2 above)
Comment (Pierre-A) [inline src/main/io/beeper.c]: this did not need to be changed anyway, residue...

**Comment by Pierre-A** [comment]
**@ledvinap**, thanks for reviewing.

**Comment by joshuabardwell** [comment]
The "preferred" and "-preferred" syntax seems confusing to me. Would "beeper save" and "beeper load" be better? Something like that?

**Comment by Pierre-A** [comment]
**@joshuabardwell**, well I want to keep the same structure and functionning for each item of the _beeper_ command (to be less confusing) and so one has to find a unique word for loading and saving this _preferred_ configuration. Would _preference_ be more explicit ? any other suggestion ?

**Comment by joshuabardwell** [comment]
It's just counter-intuitive to me that you use "-preferred" to reload the option, when the "minus" syntax is usually used to disable an option. So it seems like "-preferred" would actually un-load the preferred setting, if anything.

**Comment by Pierre-A** [comment]
**@joshuabardwell**, no, invert the way you think of it. _Prefered_ is for a subset of conditions you are interested in for disabling the beeper  in one go and _-Prefered_ allows disabling this subset as for _All_ or any other one in the list.

**Comment by hydra** [comment]
I started to read the conversation above regarding '-PREFERED' and 'PREFERED' and it made no sense to me within the couple of minuites I spent looking.

We need to expose the functionality via MSP so that we can update the GUI to enable/disable beeper settings in a user friendly way.

**Comment by hydra** [comment]
also, 'prefered' is mis-spelt.  it should be 'preferred'.

**Comment by Pierre-A** [comment]
I removed the "prefer(r)ed" condition which obviously was a bit confusing and not really essential.
The current configuration is saved in CLI and loaded when the FC is powered on.
Exposing the functionality via MSP is the next step.

**Comment by Pierre-A** [comment]
Guys, any ide

**Comment by Pierre-A** [comment]
Guys, any ideof this Tra

**Comment by Pierre-A** [comment]
Guys, any idea about this Travis failure when building Motolab target ?
(and while there's no issue with my Eclipse config).

**Comment by ledvinap** [comment]
@Pierre-A : Probably transient travis problem. Either wait until @hydra restarts it, or push new dummy commit to force restart

**Comment by Pierre-A** [comment]
thanks Petr, will wait..

**Comment by Pierre-A** [comment]
Last commit has to be checked as I am not involved in the development of Configurator.

**Comment by hydra** [comment]
great, thanks @Pierre-A

**Comment by Pierre-A** [comment]
Solve  Configurator issue #351

**Comment by Pierre-A** [comment]
**@hydra**, **@ledvinap** , any idea of what these conflicts are ?
 as this rebasing works fine locally : compiling, unittests and running after flashing on a Naze rev5.

**Comment by hydra** [comment]
not sure how you've done your rebase, i think you lost lots of commit history since your 'rebase' commit contains lots of code from master.   this PR should contain just the differences from master.

perhaps remove that last commit and try again?

usually `git rebase master` does the trick - and you stop and edit conficts as it rebases.

**Comment by hydra** [comment]
this PR contains a bad commit, I kept all the commits from your branch and then merged master into it and made appropriate changes.  likely there are some mistakes/errors but it compiles now.  Please see #2112 - if you could review the changes and test the code we can merge it into v1.13.0.

**Pull request closed without merging**